framework/src/meshgenerators/XYZDelaunayGenerator.C

```diff
@@ -201,16 +274,22 @@ XYZDelaunayGenerator::generate()
   // want to pick a new ID on hole N that doesn't conflict with any
   // IDs on hole M < N (or with the IDs on the new triangulation)
 
-  // The new triangulation by default assigns BCID i+1 to hole i ...
-  // but we can't even use this for mesh stitching, because we can't
-  // be sure it isn't also already in use on the hole's mesh and so we
-  // won't be able to safely clear it afterwards.
+  // The Netgen generated mesh does not have hole and output boundary ids,
+  // which is different from its 2D counterpart.
+  // So, we need to assign boundary ids to the hole boundaries and output boundary
+  // By default, we assign the hole boundary ids to be 1,2,..., N
```

Comment:
Why are we doing this here, though?  This is a missing feature at the libMesh level.  Would it make more sense to fix it there rather than to work around the brokenness here?

Reply:
It can use the existing algorithm you wrote here to facilitate stitching.

Reply:
The existing algorithm takes 1,2,...N defaults and remaps them, right?  The remapping part belongs here, since it has to do with the combination of stitching and triangulation, which are independent at the lower level.  But the setting of defaults is independent of stitching and ought to be done along with the triangulation.

Reply:
The existing algorithm finds the boundaries itself without taking and remapping the 1,2,...N defaults (because the defaults are never generated).

Reply:
@miaoyinb did you create a feature request on the libmesh repo for creating the default boundaries with this 1,2...N id-ing scheme?

Reply:
IMO we can adapt MOOSE later once libMesh is updated.

Reply:
please link to the issue here OR link this PR in the issue